src/main/java/frc/robot/Ports.java

```diff
@@ -11,7 +11,37 @@
     }
 
     public static class Controls {
-        public static final int XBOX = 0;
+        public static final int XBOX = 0; // Xbox controller port.
+        public static final int JOYSTICK_RIGHT = 0; // Right joystick port.
+        public static final int JOYSTICK_LEFT = 0; // Left joystick port.
     }
 
+    public static class SwerveDrive {
+        private static final int[] fl_Ports = new int[]{0, 0}; // ports for the front left motors. [drive, angle]
+        private static final int[] fr_Ports = new int[]{0, 0}; // ports for the front right motors. [drive, angle]
+        private static final int[] rl_Ports = new int[]{0, 0}; // ports for the rear left motors. [drive, angle]
+        private static final int[] rr_Ports = new int[]{0, 0}; // ports for the rear right motors. [drive, angle]
+
+        public static final int[][] motorPorts = new int[][]{ // all the ports for the motors.
+                fl_Ports, fr_Ports, rl_Ports, rr_Ports
+        };
+
+        private static final boolean[] fl_INVERTED = new boolean[]{false, false}; // inversions for the front left motors. [drive, angle]
+        private static final boolean[] fr_INVERTED = new boolean[]{false, false}; // inversions for the front right motors. [drive, angle]
+        private static final boolean[] rl_INVERTED = new boolean[]{false, false}; // inversions for the rear left motors. [drive, angle]
+        private static final boolean[] rr_INVERTED = new boolean[]{false, false}; // inversions for the rear right motors. [drive, angle]
+
+        public static final boolean[][] motor_INVERTED = new boolean[][]{ // all the inversions for the motors.
+                fl_INVERTED, fr_INVERTED, rl_INVERTED, rr_INVERTED
+        };
+
+        private static final boolean[] fl_SENSOR_PHASE = new boolean[]{false, false}; // inversions for the front left motors. [drive, angle]
+        private static final boolean[] fr_SENSOR_PHASE = new boolean[]{false, false}; // inversions for the front right motors. [drive, angle]
+        private static final boolean[] rl_SENSOR_PHASE = new boolean[]{false, false}; // inversions for the rear left motors. [drive, angle]
+        private static final boolean[] rr_SENSOR_PHASE = new boolean[]{false, false}; // inversions for the rear right motors. [drive, angle]
+
+        public static final boolean[][] motor_SENSOR_PHASE = new boolean[][]{ // all the inversions for the motors.
+                fl_SENSOR_PHASE, fr_SENSOR_PHASE, rl_SENSOR_PHASE, rr_SENSOR_PHASE
+        };
+    }
```

Comment:
Consider using [`TalonSRXConfiguration`](https://www.ctr-electronics.com/downloads/api/java/html/classcom_1_1ctre_1_1phoenix_1_1motorcontrol_1_1can_1_1_talon_s_r_x_configuration.html) instead of simple arrays.

Reply:
Only the steering motor is a TalonSRX.

Reply:
Talon SRX and Talon FX and pretty much the same, software-wise. If you'll look at the API reference you'll find a [`TalonFXConfiguration`](https://www.ctr-electronics.com/downloads/api/java/html/classcom_1_1ctre_1_1phoenix_1_1motorcontrol_1_1can_1_1_talon_f_x_configuration.html) object too...